File: src/main/java/util/Notepad/NotepadProcessor.java
package util.Notepad;

import lombok.extern.log4j.Log4j2;
import util.Platform;

import java.awt.*;
import java.io.IOException;

@Log4j2
public class NotepadProcessor {
    String platformMissing = "Unable to determine platform";
    Notepad notepad;

    public NotepadProcessor() {
        Platform p = Platform.Companion.detectPlatform();
        switch (p) {
            case WINDOWS:
                notepad = new WinNotepad();
                break;
            case LINUX:
                notepad = new LinuxNotepad();
                break;
            case MAC:
                notepad = new MacNotepad();
                break;
            case UNKNOWN:
                log.error(platformMissing);
        }
    }

    public void openNotepad() throws IOException, InterruptedException, AWTException {
        notepad.openNotepad();
    }

    public void writeText(String text) {
        notepad.writeText(text);
    }

    public void deleteText() {
        notepad.deleteText();
    }

    public void addNewLine() {
        notepad.addNewLine();
    }

    public void saveFileAs(String name) {
        notepad.saveFileAs(name);
    }

    public void openNewFile() {
        notepad.openNewFile();
    }

    public void closeFile() {
        notepad.closeFile();
    }

    public void closeNotepad() throws InterruptedException {
        notepad.closeNotepad();
    }
}

File: src/main/kotlin/util/notepad/NotepadProcessor.kt
package util.notepad

import util.emu.Platform.Companion.currentPlatform
import util.emu.Platform.LINUX
import util.emu.Platform.MAC
import util.emu.Platform.WINDOWS

class NotepadProcessor {
    private val notepad: Notepad =
        when (currentPlatform) {
            WINDOWS -> WinNotepad()
            LINUX -> LinuxNotepad()
            MAC -> MacNotepad()
            else -> error("Platform not supported")
        }

    fun openNotepad() = notepad.openNotepad()

    fun deleteText() = notepad.deleteText()

    fun addNewLine() = notepad.addNewLine()

    fun saveFileAs(name: String) = notepad.saveFileAs(name)

    fun openNewFile() = notepad.openNewFile()

    fun closeNotepad() = notepad.closeNotepad()
}
